### ISSUE TITLE
xmipp ctf selection finishes prematurely

### ISSUE THREAD

**Issue by pconesa** [I2PC/scipion#1631]
When input is still in streaming it closes and finishes while input is still running.
![image](https://user-images.githubusercontent.com/785633/40623280-560c4eac-62a6-11e8-979b-836530d7783f.png)

**Comment by pconesa** [comment]
@majtner it this something you streamified?

**Comment by majtner** [comment]
No, I was not working on this one. As far as I know, @ajimoreno was working on streaming process of this protocol.

**Comment by pconesa** [comment]
Ok, sorry! I assign this to her.

**Comment by ajimoreno** [comment]
I'm checking it, but I can't find the problem with the selection protocol. It's running correctly except when it goes after the ctf consensus. I don't know exactly the aim of the ctf consensus, but it seems it gets stuck and the output is not updated and that's the reason why ctf selection stops.

![screenshot from 2018-05-28 20-45-42](https://user-images.githubusercontent.com/26138342/40626366-609f300c-62b8-11e8-90ae-0533b5df6c1f.png)

**Comment by majtner** [comment]
It is possible that there is some problem with ctf consensus. Today, I was looking at consensus protocol and I found some scenarios, when the output was not correct. I will double-check and correct the consensus protocol, maybe it will solve this problem.

**Comment by pconesa** [comment]
Ok, thank you both!
